src/updateManager.ts

```diff
@@ -30,28 +30,38 @@ export class UpdateManager {
 
     const succeeded = await this.db.transaction(async trx => {
       const rows = await trx(this.tableName)
```

Comment:
The directus settings is a singleton, so there should just be one row, not sure what this is trying to achieve?

Reply:
This is related to the first comment, because the there is a chance that a directus_settings entry won't exist after the initialization of the project.

This code snippet assumes that a directus_settings entry exists and selects it based on the following conditions:
https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/updateManager.ts#L32-L40

If it doesn't exist, then the import process will fail.

Reply:
Right, however this would only insert values for the mv_locked/hash/ts but non of the other values. Are you calling `directus install` or `directus bootstrap` at some point to fill those in, and when they are filled in, wouldn't it create a new row?

Reply:
When launching directus via docker, `cli.js bootstrap` is executed, then `cli.js start` afterwards.

`npx directus bootstrap` and `npx directus install` both initialize the database with tables according to the yaml files in the `directus/api/src/database/seeds` directory. But `npx directus bootstrap` also migrates the database and runs this snippet of code:

https://github.com/directus/directus/blob/13cbe22d07b48f07edcfe85fad4cb8d3a524f12f/api/src/cli/commands/bootstrap/index.ts#L45-L48

As seen above, if the `PROJECT_NAME` environment variable exists, then the `settingsService.upsertSingleton()` method is executed with the appropriate argument. This method updates the `directus_settings` entry if found, otherwise, it will create one. 

I've observed that the created entry contains other values as well, and this due to setting default values in the database table itself as seen by the yaml file below:

https://github.com/directus/directus/blob/13cbe22d07b48f07edcfe85fad4cb8d3a524f12f/api/src/database/seeds/13-settings.yaml

In conclusion, the other values are automatically filled in by the database after creating a new `directus_settings` entry.